app/assets/javascripts/admin/addon/components/admin-branding-fonts-form.gjs

```diff
@@ -0,0 +1,286 @@
+import Component from "@glimmer/component";
+import { cached, tracked } from "@glimmer/tracking";
+import { fn } from "@ember/helper";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import { classify, decamelize, underscore } from "@ember/string";
+import DButton from "discourse/components/d-button";
+import Form from "discourse/components/form";
+import UpdateDefaultTextSize from "discourse/components/modal/update-default-text-size";
+import concatClass from "discourse/helpers/concat-class";
+import { ajax } from "discourse/lib/ajax";
+import { bind } from "discourse/lib/decorators";
+import { i18n } from "discourse-i18n";
+import eq from "truth-helpers/helpers/eq";
+
+const FONTS = [
+  "Helvetica",
+  "Inter",
+  "Lato",
+  "Montserrat",
+  "OpenSans",
+  "Poppins",
+  "Roboto",
+  "Merriweather",
+  "Mukta",
+];
+const MORE_FONTS = [
+  "Arial",
+  "Lora",
+  "NotoSans",
+  "NotoSansJP",
+  "Nunito",
+  "Oswald",
+  "Oxanium",
+  "PT Sans",
+  "PlayfairDisplay",
+  "Raleway",
+  "RobotoCondensed",
+  "RobotoMono",
+  "RobotoSlab",
+  "SourceSansPro",
+  "System",
+  "Ubuntu",
+];
+
+const TEXT_SIZES = ["smaller", "normal", "larger", "largest"];
+
+export default class AdminBrandingFontsForm extends Component {
+  @service siteSettings;
+  @service siteSettingChangeTracker;
+  @service toasts;
+  @service modal;
+  @service router;
+
+  @tracked moreBaseFonts = MORE_FONTS.includes(
+    classify(this.siteSettings.base_font)
+  );
+  @tracked moreHeadingFonts = MORE_FONTS.includes(
+    classify(this.siteSettings.heading_font)
+  );
+  updateExistingUsers = null;
+
+  @bind
+  setUpdateExistingUsers(value) {
+    this.updateExistingUsers = value;
+  }
+
+  @action
+  setButtonValue(fieldSet, value) {
+    fieldSet(decamelize(underscore(value)));
+  }
+
+  @action
+  toggleMoreBaseFonts() {
+    this.moreBaseFonts = !this.moreBaseFonts;
+  }
+
+  @action
+  toggleMoreHeadingFonts() {
+    this.moreHeadingFonts = !this.moreHeadingFonts;
+  }
+
+  @action
+  async update(data) {
```

Comment:
To improve the UX experience, I think we should introduce a loading state here and set the `@isLoading` attribute of `DButton` while awaiting a response from the server.

Reply:
FormKit has that fancy animation

https://github.com/user-attachments/assets/87a390f1-82c1-416b-9d56-9e15848129c7

Reply:
O nice. Is that activated as long as we return a promise?